### ISSUE TITLE
Long recordings (~30 min): long_video cluster config + CPU question

### ISSUE THREAD

**Issue by Angela680** [Zimmer-lab/wbfm#1143]
I am working on the cluster_config.yaml for long recordings. For ~30 min recordings, I plan to use a 5-day controller instead of 3 days and extend the walltimes for the steps that need longer processing.

Question about cpus_per_task

While troubleshooting, I also looked at the Slurm accounting for several behavior rules. In the runs I checked:

sam2_segment: requested 8 CPUs, averaged ~1 CPU
coil_unet: requested 8 CPUs, averaged ~1.1 CPU
normalize_img: requested 8 CPUs, averaged ~1 CPU
subtract_background: requested 8 CPUs, averaged ~1 CPU
create_centerline: requested 2 CPUs, averaged ~1 CPU
binarize_coil: requested 2 CPUs, averaged ~1 CPU

For binarize_coil and create_centerline, total CPU time was also quite close to wall-clock time, so I wondered whether they were mostly using one CPU in these runs.

I also had a look at the corresponding code paths, although I may be missing something in how the underlying libraries use threads. For create_centerline, binarize_coil, normalize_img, and subtract_background, I did not see an obvious multiprocessing/thread-pool mechanism, and I also did not find an obvious worker pool in the sam2_segment wrapper.

Am I interpreting this correctly? Would it be worth to set lower cpus_per_task values for these rules?

**Comment by Charles-Fieseler-Vienna** [comment]
This is really good detective work! In principle a lot of the multithreading happens in pytorch/tensorflow, so there isn't any thread dedicated code in my python. Going down to 2 CPUs for several of these is good; I wouldn't go down to 1 just in case

**Comment by Charles-Fieseler-Vienna** [comment]
Some strange things:
1. I have sam2 requesting 24 CPUs, not 8
2. Was coil_unet that low when gpu was working? Perhaps we should decrease the cpu request and just crash if it doesn't find gpu

Otherwise I decreased everything

Does this have

**Comment by Angela680** [comment]
1. sam2_segment

Ah yes, I had already changed sam2_segment from 24 CPUs to 8 earlier, probably after the LiSC email/recommendation...

2. coil_unet

Yes, I checked the specific coil_unet job. The GPU was definitely working: TensorFlow created GPU:0 on an NVIDIA L40S, loaded cuDNN 8700, and reported the GPU in tf.config.list_physical_devices('GPU').

That job had 8 CPUs allocated, but TotalCPU was 16:19:24 for 16:02:55 elapsed, so it averaged about 1 CPU while the GPU was working.

**Comment by Charles-Fieseler-Vienna** [comment]
Sounds good!
